daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -886,6 +886,1034 @@ def make_compute(sdfg, state, vec_width=1):
+        new_sdfg.add_datadesc("Y", copy.deepcopy(Y))
+
+        new_sdfg.arrays["X"].transient = False
+        new_sdfg.arrays["W"].transient = False
+        new_sdfg.arrays["Y"].transient = False
+
+        if B is not None:
+            new_sdfg.add_datadesc("B", copy.deepcopy(B))
+            new_sdfg.arrays["B"].transient = False
+
+        # GEMM Parameters
+        vec_width = Y.dtype.veclen
+        vec_width_in = X.dtype.veclen
+        base_type = Y.dtype.base_type
+        vec_type = dace.vector(base_type, vec_width)
+
+        # matrix dimensions
+        K = num_channels * filter_hx * filter_hy
+        M = output_size_y * vec_width * output_size_x  # account for vectorization on Y
+        N = num_filters
+
+        # Set number of processing elements
+        P = num_filters if pe is None else pe
+
+        # default, set tile size to full image (i.e. no tiling)
+        if tiles is None:
+            tiles = M
+
+        # tiles not larger than entire image and at least vector size
+        tile_size_m = min(tiles, M)
+        if vec_width > tile_size_m:
+            tile_size_m = vec_width
+
+        # veclen and tile size need to be compatible
+        if not tile_size_m % vec_width == 0:
+            print(
+                f"Given vector length and tile size not compatible ({tile_size_m}, {vec_width}), reset to",
+                end="")
+            tile_size_m = np.lcm(tile_size_m, vec_width)
+            print(f"{tile_size_m}")
+
+        # fix tile size
+        T = tile_size_m
+
+        if T % output_size_x != 0:
+            raise NotImplementedError(
+                f"Currently tile size ({tile_size_m}) must \
+                be a multiple of the output image width {output_size_x}")
+
+        # Safe delay: this will be used in the compute state, pipeline scope, to insert
+        # a delay between accumulation on the same result if needed.
+        # Further explanations are provided in the compute state.
+
+        # Note: this is a platform and type dependent parameter.
+        L = max(16 - T / vec_width, 0)
+
+        # This implementation uses a flattened nested loop, that overlaps feeding,
+        # computing and draining phases. Each PE is responsible for computing one
+        # tile of one row of the final result C. With the current implementation,
+        # A PE needs K*T cycles to compute the results and then P*T clock cycles
+        # to fully drain them (draining is distributed across PEs).
+        # Therefore, in order to guarantee correctness and deadlock free we have
+        # to ensure that the number of cycles needed to drain the results is less
+        # or equal to the number of cycles needed to compute them.
+        # That is PT <= KT.
+        if K is not None and P > K:
+            raise ValueError(
+                f"Conv Im2Col (tiled): Number of processing elements {P} must be smaller than the K-dimension {K}."
+            )
+
+        def make_read_A(state):
```

Comment:
I think it makes more sense to name it `make_read_W` and mention that this is the `A` of the classical GEMM
(and update the following comments as well)

Reply:
addressed